WordPress/gutenberg
# Navigation Block: Add tests for Nav block uncontrolled blocks dirty state checking

## What?


Adds unit tests for work done in https://github.com/WordPress/gutenberg/pull/46223 and improves function naming.

## Why?


This is a fragile point in the code and thus deserves some unit tests at least.

Also helps make the React component responsible for fewer things.

## How?


- Extracts function and renames to be more generic.
- Adds unit tests

## Testing Instructions





Run through test steps in https://github.com/WordPress/gutenberg/pull/46223 to check for regressions.

Also run tests

```
npm run test:unit edit/test/are-blocks-dirty.js
```

### Testing Instructions for Keyboard


## Screenshots or screencast
